apollo-router/src/query_planner/mod.rs

```diff
@@ -478,11 +457,11 @@ impl PlanNode {
                         let subselection = deferred_node.subselection();
                         let label = deferred_node.label.clone();
                         let mut tx = sender.clone();
-                        let sc = schema.clone();
-                        let orig = originating_request.clone();
-                        let sf = service_factory.clone();
-                        let ctx = context.clone();
-                        let opt = options.clone();
+                        let sc = parameters.schema.clone();
+                        let orig = parameters.originating_request.clone();
+                        let sf = parameters.service_factory.clone();
+                        let ctx = parameters.context.clone();
+                        let opt = parameters.options.clone();
```

Comment:
I think we can destructure here ?

Reply:
If destructured we would only get references, they still have to be cloned to get in the deferred node's future

Reply:
true